File: app/src/main/java/com/haoyu/app/adapter/CoachCourseAdapter.java
//package com.haoyu.app.adapter;
//
//import android.content.Context;
//import android.view.View;
//import android.widget.ImageView;
//import android.widget.LinearLayout;
//import android.widget.TextView;
//
//import com.haoyu.app.basehelper.BaseArrayRecyclerAdapter;
//import com.haoyu.app.entity.CourseMobileEntity;
//import com.haoyu.app.entity.TimePeriod;
//import com.haoyu.app.imageloader.GlideImgManager;
//import com.haoyu.app.lego.teach.R;
//import com.haoyu.app.utils.PixelFormat;
//import com.haoyu.app.utils.ScreenUtils;
//import com.haoyu.app.utils.TimeUtil;
//
//import java.util.List;
//
///**
// * 创建日期：2017/12/15.
// * 描述:我参与辅导的课程  首页课程列表
// * 作者:xiaoma
// */
//
//public class CoachCourseAdapter extends BaseArrayRecyclerAdapter<CourseMobileEntity> {
//    private Context context;
//    private int imageWidth;
//    private int imageHeight;
//
//    public CoachCourseAdapter(Context context, List<CourseMobileEntity> mDatas) {
//        super(mDatas);
//        this.context = context;
//        imageWidth = ScreenUtils.getScreenWidth(context) / 3 - PixelFormat.formatPxToDip(context, 20);
//        imageHeight = imageWidth / 3 * 2;
//    }
//
//    @Override
//    public void onBindHoder(RecyclerHolder holder, final CourseMobileEntity entity, int position) {
//        LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(
//                imageWidth, imageHeight);
//        ImageView course_img = holder.obtainView(R.id.course_img);
//        TextView course_title = holder.obtainView(R.id.course_title);
//        TextView course_code = holder.obtainView(R.id.course_code);
//        TextView course_period = holder.obtainView(R.id.course_period);
//        TextView course_state = holder.obtainView(R.id.course_state);
//        View line = holder.obtainView(R.id.line);
//        course_img.setLayoutParams(params);
//        GlideImgManager.loadImage(context, entity.getImage(), R.drawable.app_default, R.drawable.app_default, course_img);
//        course_title.setText(entity.getTitle());
//        String codeStr = "";
//        if (entity.getCode() != null && entity.getCode().trim().length() > 0) {
//            codeStr += entity.getCode();
//        }
//        if (entity.getTermNo() != null && entity.getCode().trim().length() > 0) {
//            codeStr += "/" + entity.getTermNo();
//        }
//        course_code.setText(codeStr);
//        TimePeriod timePeriod = entity.getmTimePeriod();
//        String text_period = "开课：";
//        if (timePeriod != null) {
//            text_period += TimeUtil.getSlashDate(entity.getmTimePeriod().getStartTime());
//            course_period.setText("开课：" + TimeUtil.getSlashDate(entity.getmTimePeriod().getStartTime()));
//        }
//        course_period.setText(text_period);
//        if (timePeriod != null && timePeriod.getMinutes() > 0) {
//            course_state.setText("进行中");
//        } else {
//            if (timePeriod != null && timePeriod.getState() != null) {
//                course_state.setText(timePeriod.getState());
//            } else {
//                course_state.setText("进行中");
//            }
//        }
//        if (position == getItemCount() - 1) {
//            line.setVisibility(View.GONE);
//        } else {
//            line.setVisibility(View.VISIBLE);
//        }
//    }
//
//    @Override
//    public int bindView(int viewtype) {
//        return R.layout.coachcourse_item;
//    }
//}

File: app/src/main/java/com/haoyu/app/adapter/CoachCourseAdapter.kt
package com.haoyu.app.adapter

import android.content.Context
import android.view.View
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import com.haoyu.app.basehelper.BaseArrayRecyclerAdapter
import com.haoyu.app.entity.CourseMobileEntity
import com.haoyu.app.imageloader.GlideImgManager
import com.haoyu.app.lego.teach.R
import com.haoyu.app.utils.PixelFormat
import com.haoyu.app.utils.ScreenUtils
import com.haoyu.app.utils.TimeUtil

/**
 * 创建日期：2018/1/24.
 * 描述:参与辅导的课程  首页课程列表
 * 作者:xiaoma
 */
class CoachCourseAdapter(private val context: Context, mDatas: List<CourseMobileEntity>) : BaseArrayRecyclerAdapter<CourseMobileEntity>(mDatas) {
    private var imageWidth = 0
    private var imageHeight = 0

    init {
        imageWidth = ScreenUtils.getScreenWidth(context) / 3 - PixelFormat.formatPxToDip(context, 20)
        imageHeight = imageWidth / 3 * 2
    }

    override fun bindView(viewtype: Int): Int {
        return R.layout.coachcourse_item
    }

    override fun onBindHoder(holder: RecyclerHolder, entity: CourseMobileEntity, position: Int) {
        val ivImage = holder.obtainView<ImageView>(R.id.ivImage)
        val tvTitle = holder.obtainView<TextView>(R.id.tvTitle)
        val tvCode = holder.obtainView<TextView>(R.id.tvCode)
        val tvPeriod = holder.obtainView<TextView>(R.id.tvPeriod)
        val tvState = holder.obtainView<TextView>(R.id.tvState)
        val line = holder.obtainView<View>(R.id.line)
        ivImage.layoutParams = LinearLayout.LayoutParams(imageWidth, imageHeight)
        GlideImgManager.loadImage(context, entity.image, R.drawable.app_default, R.drawable.app_default, ivImage)
        tvTitle.text = entity.title
        var code = ""
        entity.code?.let { code += it }
        entity.termNo?.let { code += "/$it" }
        tvCode.text = code
        var textPeriod = "开课："
        val state = if (entity.getmTimePeriod() != null) {
            entity.getmTimePeriod().let {
                textPeriod += TimeUtil.getSlashDate(it.startTime)
                if (it.state != null) {
                    it.state
                } else {
                    if (it.minutes > 0) "进行中" else ""
                }
            }
        } else "进行中"
        tvPeriod.text = textPeriod
        tvState.text = state
        if (position == itemCount - 1) {
            line.visibility = View.GONE
        } else {
            line.visibility = View.VISIBLE
        }
    }
}